File: src/ads_runner_client/lib/domain/entities/campaign.dart
import 'package:equatable/equatable.dart';
import '../../core/enums/campaign_status.dart';
import '../../core/enums/region_type.dart';
import 'media_asset.dart';

class Campaign extends Equatable {
  final String id;
  final String name;
  final String description;
  final CampaignStatus status;
  final List<MediaAsset> mediaAssets;
  final List<RegionType> targetRegions;
  final DateTime startDate;
  final DateTime endDate;
  final double budget;
  final int impressions;
  final int reach;
  final String? thumbnailUrl;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Campaign({
    required this.id,
    required this.name,
    required this.description,
    required this.status,
    required this.mediaAssets,
    required this.targetRegions,
    required this.startDate,
    required this.endDate,
    required this.budget,
    required this.impressions,
    required this.reach,
    this.thumbnailUrl,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [id, name, status, startDate, endDate, impressions, reach];
}

File: src/AdsRunner.Domain/Entities/Campaign.cs
using AdsRunner.Domain.Common;
using AdsRunner.Domain.Enums;

namespace AdsRunner.Domain.Entities;

public class Campaign : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public CampaignStatus Status { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public decimal Budget { get; set; }
    public int Impressions { get; set; }
    public int Reach { get; set; }
    public string? ThumbnailUrl { get; set; }

    // Foreign keys
    public Guid UserId { get; set; }

    // Navigation properties
    public User User { get; set; } = null!;
    public ICollection<MediaAsset> MediaAssets { get; set; } = new List<MediaAsset>();
    public ICollection<CampaignRegion> CampaignRegions { get; set; } = new List<CampaignRegion>();
    public ICollection<CampaignScreen> CampaignScreens { get; set; } = new List<CampaignScreen>();
}
